### DISCUSSION TITLE
Azure DevOps -> Pass Build.Id Parameter

### DISCUSSION THREAD

**Discussion by edwards2018** [rfennell/AzurePipelines#1041] [Q&A] [answered]
Hi I have been experimenting with the extension today, and it looks really promising in solving our release note generation problem. What I am asking might be possible already in YAML markup but its not clear how. 

Is it possible to input the Build.ID of a previous build and for the task to gather all PR/WI/CS from that Build.Id to current?

This is the extract of my YAML build pipeline so far

  - task: XplatGenerateReleaseNotes@3
    inputs:
      outputfile: '$(Build.ArtifactStagingDirectory)/releasenotes.md'
      templateLocation: 'File'
      templatefile: '$(System.DefaultWorkingDirectory)/Deployment/Deployment/ReleaseNotesTemplate/Release_Note_Template.md'
      dumpPayloadToConsole: false
      dumpPayloadToFile: false
      replaceFile: True
      getParentsAndChildren: False
      getAllParents: False
      getIndirectPullRequests: False
      stopOnError: False

**Comment by rfennell** [comment, marked as answer]
Yes it is, it is the 'OverrideBuildReleaseId' parameter. The details are in the [wiki](https://github.com/rfennell/AzurePipelines/wiki/GenerateReleaseNotes---Node-based-Cross-Platform-Task)

**Reply by edwards2018** [reply]
Thanks that sounds exactly like what I am after, apologies if these seem like basic queries, we have number of classic and YAML pipelines I'd like to integrate this into. I have successfully used the TAG range, on a YAML pipeline. 

But the OverrideBuildReleaseId I believe is applicable only at release deployment stage, is that correct or am I missing something?

**Reply by rfennell** [reply]
No, it works....

For releases or multi-stage YAML this parameter provides a means to set the ID of the 'last good release' to compare against. If the specified release/build is not found then the task will exit with an error. The override behaviour is as follows.
- (Default) Parameter undefined - Old behaviour, looks for last successful build using optional stage and tag filters
- Empty string - Old behaviour, looks for last successful build using optional stage and tag filters
- A valid build ID (int) - Use the build ID as the comparison
- An invalid build ID (int) - If a valid build cannot be found then the task exits with a message
- Any other non empty input value - The task exits with an error message

**Reply by edwards2018** [reply]
Thank you so much for the fast responses, I will give the above a go, but I think I understand where I was going wrong now. Thanks again
